pypesto/store/read_from_hdf5.py

```diff
@@ -111,13 +111,21 @@ def read(self, objective: ObjectiveBase = None) -> Problem:
         problem:
             A problem instance with all attributes read in.
         """
+        # raise warning that problem is not to be used but
```

Comment:
only makes sense to say something if objective is None.

Reply:
why only then? If we have e.g. an amici objective, we cannot load this by loading the problem from hdf5 don't we?

Reply:
sorry, I don't get it :sweat_smile:

Reply:
Well the info/warning was meant to tell the user that he cannot use the problem loaded from the hdf5 file for optimization etc. The reason for this is (as far as I understand that is the only reason) that the objective is not loaded correctly (and is also not intended to be loaded). Therefore the warning says to separately create the objective function (or the problem in general) if one aims to do optimization.

Reply:
yes, but `if objective is not None`, then everything should be fine, right? As then the user has provided an objective.

But there may be a problem as this objective is likely not accounting for fixed parameters yet, therefore maybe one should call problem.objective.update_from_problem() afterwards ...

Reply:
ahhh yeah I think now I get what you meant :D you are right 👍